### PR TITLE
Fixed all broken links of ARCHITECTURE.adoc and a faulty file name

### PR DESCRIPTION
**Pull request by maulikpareek**
### Description
-Fixed all the links to the .java files that were broken previously (Showed '404 page not found').
-There was no such file as 'ScoreEngine.java', It seems it was referring to 'ScoringEngine.java' which is present in repository. So did the correction on that as well.

- [X] The description of this pull request is detailed and explain why this pull request is needed
- [X] The changeset is on a specific branch. Using `feature/` for new feature, or improvements ; Using `fix/` for bug fixes ; Using `docs/` for any documentation changes.
- [X] If required, the documentation has been updated

### CODE REVIEW

File: docs/ARCHITECTURE.adoc [hunk 1]
```diff
@@ -1,11 +1,11 @@
 == Application architecture
 
-The application is divided into two main parts: link:../src/main/java/io/jenkins/pluginhealth/scoring/probes/Probe.java[`Probe`] and link:../src/main/java/io/jenkins/pluginhealth/scoring/scores/Scoring.java[`Scoring`].
+The application is divided into two main parts: link:/core/src/main/java/io/jenkins/pluginhealth/scoring/probes/Probe.java[`Probe`] and link:/core/src/main/java/io/jenkins/pluginhealth/scoring/scores/Scoring.java[`Scoring`].
 
 === What is a `Probe`?
 
 Probes have the responsibility to fetch data for each plugins.
-They reach external system, like GitHub, the Jira, etc, and register the wanted data into a link:../src/main/java/io/jenkins/pluginhealth/scoring/model/ProbeResult.java[`ProbeResult`].
+They reach external system, like GitHub, the Jira, etc, and register the wanted data into a link:/core/src/main/java/io/jenkins/pluginhealth/scoring/model/ProbeResult.java[`ProbeResult`].
```
Comment (alecharp) [inline docs/ARCHITECTURE.adoc:8]: This is not fixing the problem in the preview.
I think it needs to be `../core/`

**Review by alecharp — CHANGES_REQUESTED** [review]

File: docs/ARCHITECTURE.adoc (hunk 1 above)
Comment (maulikpareek) [reply docs/ARCHITECTURE.adoc:8]: But it seems to work pretty fine in my forked repository.
Kindly have a look and lemme know If I still need to change it.
https://github.com/maulikpareek/plugin-health-scoring/tree/docs

**Comment by maulikpareek** [comment]
@alecharp are changes still required?

File: docs/ARCHITECTURE.adoc (hunk 1 above)
Comment (Jagrutiti) [reply docs/ARCHITECTURE.adoc:8]: GitHub does not parse the `.adoc` correctly. If you test your syntax in the local it won't work.

Comment (maulikpareek) [reply docs/ARCHITECTURE.adoc]: alright! fixed it with ../core

**Review by Jagrutiti — APPROVED** [review]
LGTM

**Comment by maulikpareek** [comment]
@alecharp is there any problem in it? I'll fix it

**Review by alecharp — APPROVED** [review]
Sorry, I was off for the end of year celebrations. This looks good to me, thanks.

**Pull request merged**